### PR TITLE
Support Triggered Calibration as ROS2 Action

### PR DESCRIPTION
**Pull request by SamerKhshiboun**
Tracked on RSDEV-2204

Based on:
https://design.ros2.org/articles/actions.html
https://docs.ros.org/en/jazzy/Tutorials/Intermediate/Writing-an-Action-Server-Client/Cpp.html#writing-an-action-server

- Added new action TriggeredCalibration
- Updated README
- Created new actions.cpp file and implemented the action in it
- Added documentation.
- Testing will be done in RSDEV-2298

### CODE REVIEW

File: realsense2_camera/src/actions.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Intel Corporation. All Rights Reserved.
```
Comment (Nir-Az) [inline realsense2_camera/src/actions.cpp:1]: actions.cpp is a generic name.
Do we want a calibration action dedicated name instead ?

Comment (SamerKhshiboun) [reply realsense2_camera/src/actions.cpp:1]: I think we will add all actions in the future inside it (we can add FW-update action for example, etc..)
(In the same way, I will create a "services.cpp" file for all services - we can redesign this in a separate task)

File: realsense2_camera/src/actions.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+ * A callback that accepts a new goal (request) and starts processing it.
+ * Since the execution is a long-running operation, we spawn off a
+ * thread to do the actual work and return from handle_accepted quickly.
+*/
+void BaseRealSenseNode::TriggeredCalibrationHandleAccepted(const std::shared_ptr<GoalHandleTriggeredCalibration> goal_handle)
+{
+    using namespace std::placeholders;
+    ROS_INFO("TriggeredCalibrationAction: Request accepted");
+    // this needs to return quickly to avoid blocking the executor, so spin up a new thread
+    std::thread{std::bind(&BaseRealSenseNode::TriggeredCalibrationExecute, this, _1), goal_handle}.detach();
```
Comment (Nir-Az) [inline realsense2_camera/src/actions.cpp:71]: Why not opening a new block for the {}?
Do we need to worry about the detach part?
Join on destruction or something?

**Comment by Nir-Az** [comment]
Looks great overall :)

File: realsense2_camera/src/actions.cpp (hunk 2 above)
Comment (SamerKhshiboun) [reply realsense2_camera/src/actions.cpp:71]: As I saw in the formal examples in ROS2 documentation, this is enough (links attached in the PR discerption)
If you think there is a better way to maintain this, we can discuss this offline

File: realsense2_camera/src/actions.cpp (hunk 1 above)
Comment (Nir-Az) [reply realsense2_camera/src/actions.cpp:1]: On next action/service please consider creating a service/action folder and implement a file for each serve/action
With time it may grow..

**Review by Nir-Az — APPROVED** [review]
